Tests/MongoSwiftTests/UnifiedTestRunner/UnifiedTestRunner.swift

```diff
@@ -175,46 +197,47 @@ struct UnifiedTestRunner {
                     entities: try file.createEntities?.toEntityMap() ?? [:],
                     internalClient: self.internalClient
                 )
-
                 // Workaround for SERVER-39704:  a test runners MUST execute a non-transactional distinct command on
                 // each mongos server before running any test that might execute distinct within a transaction. To ease
                 // the implementation, test runners MAY execute distinct before every test.
                 if self.topologyType.isSharded && !MongoSwiftTestCase.serverless {
                     let collEntities = context.entities.values.compactMap { try? $0.asCollection() }
-                    for (_, client) in try self.internalClient.asMongosClients() {
+                    let clientMap = try self.internalClient.asMongosClients()
+                    for (_, client) in clientMap {
                         for entity in collEntities {
-                            _ = try client.db(entity.namespace.db).runCommand(
-                                ["distinct": .string(entity.name), "key": "_id"]
-                            )
+                            do {
+                                _ = try await client.db(entity.namespace.db).runCommand(
+                                    ["distinct": .string(entity.name), "key": "_id"]
+                                )
+                            } catch {
+                                // DB not explicitly initially created for operation-failure test bc no initialData
```

Comment:
can we make this comment a little mores pecific? something like "this command can fail for tests that have no `initialData` specified and so do not create the namespace, such as the `valid-fail/operation-failure.json` test"